src/main/scala/Board.scala

```diff
@@ -133,6 +121,45 @@ case class Board(
 
   def materialImbalance: Int = variant.materialImbalance(this)
 
+  lazy val kingsAndBishopsOnly: Boolean =
+    (kings | bishops) == occupied
+
+  lazy val kingsAndKnightsOnly: Boolean =
+    (kings | knights) == occupied
+
+  lazy val onlyKnights: Boolean = knights == occupied
+
+  lazy val minors: Bitboard =
+    bishops | knights
+
+  lazy val kingsAndMinorsOnly: Boolean =
+    (kings | minors) == occupied
+
+  lazy val kingsRooksAndMinorsOnly: Boolean =
+    (kings | rooks | minors) == occupied
```

Comment:
`lazy val` has a runtime cost. Is it lower than the computation it caches in the example above? I'm not sure.

Reply:
maybe they should be just def. We use them once or twice anyway. I'll take note.